src/core/Slider/Slider.tsx

```diff
@@ -0,0 +1,638 @@
+      } else if (event.key === 'End') {
+        setThumbValue(index, maxVal);
+      }
+    },
+    [thumbMode, values, index, railDomain, step, setThumbValue],
+  );
+
+  const handlePointerDownOnThumb = React.useCallback(
+    (event: React.PointerEvent) => {
+      thumbRef.current?.focus();
+      setThumbMoveIndex(index);
+      event.preventDefault();
+      event.stopPropagation();
+    },
+    [index, setThumbMoveIndex],
+  );
+
+  const [hasFocus, setHasFocus] = React.useState(false);
+  const [isHovered, setIsHovered] = React.useState(false);
+
+  const generateTooltip = React.useCallback(
+    (val: number): string => {
+      return toolTipFunction
+        ? toolTipFunction(val, step)
+        : formatNumberValue(val, step);
+    },
+    [toolTipFunction, step],
+  );
+
+  const currentValue = values[index];
+  const tooltipContent = generateTooltip(currentValue);
+  const leftPercent =
+    (100.0 * (currentValue - railDomain.min)) /
+    (railDomain.max - railDomain.min);
+
+  const thumbClassNames = React.useMemo(
+    () => cx('iui-slider-thumb', isActive && 'iui-active'),
+    [isActive],
+  );
+  const showTooltip = !hideTooltip && (isActive || hasFocus || isHovered);
+  return (
+    <Tooltip
+      visible={showTooltip}
+      content={tooltipContent}
+      placement={toolTipPosition}
+    >
+      <div
+        ref={thumbRef}
+        style={{ left: `${leftPercent}%` }}
+        className={thumbClassNames}
+        role='slider'
+        tabIndex={0}
+        aria-valuemin={railDomain.min}
+        aria-valuenow={railDomain.max}
+        aria-valuemax={currentValue}
+        onPointerDown={handlePointerDownOnThumb}
+        onKeyDown={handleOnKeyDown}
+        onFocus={() => setHasFocus(true)}
+        onBlur={() => setHasFocus(false)}
+        onMouseEnter={() => setIsHovered(true)}
+        onMouseLeave={() => setIsHovered(false)}
+      />
+    </Tooltip>
+  );
+};
+
+/** Properties for [[Slider]] component */
+export type SliderProps = {
+  /**
+   * Set focus on first thumb in slider element.
+   * @default false
+   */
+  setFocus?: boolean;
+  /**
+   * Allowable range of values.
+   */
+  railDomain: ValueDomain;
+  /**
+   * Array of one or more values to show.
+   */
+  values: number[];
+  /**
+   *  Determines which segments are shown with color.
+   * 'none' - no colored tracks are displayed.
+   * 'odd-segments'- colored tracks shown in segments 1,3,5, etc. Default if number of thumbs values are odd.
+   * 'even-segments'- colored tracks shown in segments 2,4,6, etc. Default if number of thumbs values are even.
+   * @default undefined
+   */
+  trackDisplayMode?: TrackDisplayMode;
+  /**
+   * Step increment controls what values are allowed and the amount the value will change when left and
+   * right arrows are pressed when a Thumb has focus.
+   * @default 1
+   */
+  step?: number;
+  /** Forces control to be displayed in a disabled state where no interactive value changes are allowed.
```

Comment:
new line

Reply:
This newline issue is everywhere in this file. Please read our style guide

https://github.com/iTwin/iTwinUI-react/blob/cec7085c269b2f929bb0fd4b052cf47e569a9594/STYLEGUIDE.md#L47-L49